iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -244,102 +254,96 @@ Viewer.prototype._addIdToNode = function(node, id) {
+    const isFact = (name == 'NONNUMERIC' || name == 'NONFRACTION');
+    if (n.nodeType == 1) {
+        // Ignore iXBRL elements that are not in the default target document, as
+        // the viewer builder does not handle these, and the builder does not
+        // ensure that they have ID attributes.
         const id = n.getAttribute("id");
-        if (inHidden) {
-            nodes = $(n);
-        } else {
-            nodes = this._findOrCreateWrapperNode(n);
-        }
-        this._addIdToNode(nodes.first(), id);
-        /* We may have already created an IXNode for this ID from a -sec-ix-hidden
-         * element */
-        var ixn = this._ixNodeMap[id];
-        if (!ixn) {
-            ixn = new IXNode(id, nodes, docIndex);
-            this._ixNodeMap[id] = ixn;
-        }
-        if (nodes.is(':hidden')) {
-            ixn.htmlHidden = true;
-        }
-        if (inHidden) {
-            ixn.isHidden = true;
-            nodes.addClass("ixbrl-element-hidden");
-        }
-        if (n.getAttribute("continuedAt")) {
-            this._continuedAtMap[id] = { 
-                "isFact": name != 'CONTINUATION',
-                "continuedAt": n.getAttribute("continuedAt")
-            }
-        }
-        if (name == 'CONTINUATION') {
-            $(nodes).addClass("ixbrl-continuation");
-            ixn.isContinuation = true;
-        }
-        else {
-            this._docOrderIDIndex.push(id);
-        }
-        if (name == 'NONFRACTION') {
-            $(nodes).addClass("ixbrl-element-nonfraction");
-        }
-        if (name == 'NONNUMERIC') {
-            $(nodes).addClass("ixbrl-element-nonnumeric");
-            if (n.hasAttribute('escape') && n.getAttribute('escape').match(/^(true|1)$/)) {
-                ixn.escaped = true;
+        if (((isFact || isFootnote) && !n.hasAttribute("target"))
+            || (isContinuation && this.continuationOfMap[id] !== undefined)) {
+            var nodes;
+            if (inHidden) {
+                nodes = $(n);
+            } else {
+                nodes = this._findOrCreateWrapperNode(n);
             }
-        }
-        if (isFootnote) {
-            $(nodes).addClass("ixbrl-element-footnote");
-            ixn.footnote = true;
-        }
-    }
-    else if(n.nodeType == 1 && name == 'HIDDEN') {
-        inHidden = true;
-    }
-    else if(n.nodeType == 1) {
-        // Handle SEC/ESEF links-to-hidden
-        const id = this._getIXHiddenLinkStyle(n);
-        if (id !== null) {
-            nodes = $(n);
-            nodes.addClass("ixbrl-element").data('ivid', [id]);
-            this._docOrderIDIndex.push(id);
-            /* We may have already seen the corresponding ix element in the hidden
-             * section */
+
+            // For a continuation, store the IX ID of the item, not the continuation
+            const headId = isContinuation ? this.continuationOfMap[id] : id;
+            this._addIdToNode(nodes.first(), headId);
```

Comment:
It looks like `continuationOfMap` is an map<continuationId,list<primaryId>>, so `headId` is now either a single id or an array, but it looks like `_addIdToNode` wants a single id.  Is this working as expected?

Reply:
Good spot.  Amazingly, this does appear to be working as intended (by fluke), because apparently in JS:

```javascript
> arr = ["a","b","c"];
(3) ['a', 'b', 'c']
> arr[1];
'b'
> arr[[1]];
'b'
```

I've no idea why this array flattening happens, but have fixed this code in 1adcede.

Reply:
I know keys are `.toString()`'ed before being used as keys, but presumably if there ever _were_ multiple keys that things wouldn't work?  I guess the scenario is two fact targeting the same continuation.  If we don't already have a test for that (e.g., manual), it might be valuable to add one.

Reply:
Yes, I should have said "it's not as obviously broken as you might expect" :-)  

This is intended to deal with a continuation that is shared between two facts, but it turns out that that's actually [illegal](https://www.xbrl.org/specification/inlinexbrl-part1/rec-2013-11-18/inlinexbrl-part1-rec-2013-11-18.html#d1e1266):

> If the continuedAt attribute is present, the {continuation reference} property MUST NOT match the {continuation reference} property of any other element in the [Inline XBRL Document Set](https://www.xbrl.org/specification/inlinexbrl-part1/rec-2013-11-18/inlinexbrl-part1-rec-2013-11-18.html#term-inlinexbrldocumentset).

I'll remove this, and replace it with a check that we don't have shared continuations.

Reply:
Done in ff4f118